storage/storage_image.go

```diff
@@ -93,8 +94,14 @@ func newImageSource(ctx context.Context, sys *types.SystemContext, imageRef stor
 		return nil, err
 	}
 
+	directory, err := ioutil.TempDir(tmpdir.TemporaryDirectoryForBigFiles(sys), "storage")
```

Comment:
This is never removed? Does it even need to exist?

Reply:
I'll check where we could remove it.

I think it needs to exist here, since GetBlob doesn't have a `SystemContext` that we need for the big temp dir.

Reply:
`storageImageSource` could just hold a reference to the `SystemContext` directly.

Reply:
Even better, thanks :)